### ISSUE TITLE
Climate data - Temperature

### ISSUE THREAD

**Issue by Mohinem** [FAIRiCUBE/data-requests#32]

**Comment by KathiSchleidt** [comment]
Mohit is struggling to understand which of the meteorological variables available from the provided data source are required by the UC. Adding UC responsibles for their feedback

**Comment by robknapen** [comment]
Which data source is this about?

**Comment by robknapen** [comment]
Thanks @Mohinem. From that all variables look interesting to me. But I don't think we need hourly data. Daily min, max and mean values would be ok. Starting from the year 2000.

For our use case we can start with altitude, near-surface air temperature, near-surface humidity, rainfall flux, and snowfall flux.

I guess if needed other data can still be ingested later?

**Comment by Mohinem** [comment]
That makes sense. I guess I will import all the variables then. And yest, other data can be ingested later.

**Comment by Mohinem** [comment]
> Thanks @Mohinem. From that all variables look interesting to me. But I don't think we need hourly data. Daily min, max and mean values would be ok. Starting from the year 2000.
> 
> For our use case we can start with altitude, near-surface air temperature, near-surface humidity, rainfall flux, and snowfall flux.
> 
> I guess if needed other data can still be ingested later?

Rob, the dataset only has monthly data. So do you mean the dataset should be in such a way that it's easy to derive monthly min, max and mean values ?

**Comment by robknapen** [comment]
Hi @Mohinem , I am not that familiar with this particular dataset. However, I can imagine that for our use case weekly or monthly climate data would be sufficient. Usually for rainfall and snowfall this means calculating the cumulative value over the period (so e.g. total rainfall per month). The others can be mean, stddev, min and max values.

**Comment by KathiSchleidt** [comment]
@Mohinem looking at the overview page, I see `Temporal resolution: Hourly`, could you please check this?

@robknapen @capoony @sonjastndl @heimor 
looking at the inventory sheet, I see only 3 of the variables from this source requested (Temperature, Precipitation, Wind), when I look at the source description, I see more:

- **Near-surface air temperature**
- Near-surface specific humidity
- **Near-surface wind speed**
- _Rainfall flux_
- _Snowfall flux_
- Surface air pressure
- Surface downwelling longwave radiation
- Surface downwelling shortwave radiation

I'm assuming that the request for precipitation should actually refer to the Rainfall and Snowfall Flux variables. In the description is seems that these are derived from a more general ERA5 total precipitation source. Should we look for a more general "Precipitation" dataset, or are you OK with this split into Rain and Snow?

Also - the other variables also seem relevant, would you need these ingested?

**Comment by robknapen** [comment]
@KathiSchleidt Please see my previous comment about the variables: https://github.com/FAIRiCUBE/data-requests/issues/32#issuecomment-1600816103

For UC2 purposes I think it is fine to sum daily, weekly, or monthly cumulative rainfall and snowfall to get an idea of total precipitation.

**Comment by heimor** [comment]
dear Kathi,


in principal all variables are relevant,


importance descending


  *   Rainfall flux
  *
  *   Near-surface specific humidity
  *   Near-surface air temperature
  *
  *   Near-surface wind speed


  *
  *   Snowfall flux


  *
  *   Surface downwelling longwave radiation
  *   Surface downwelling shortwave radiation
  *   Surface air pressur


generally cumulative monthly rainfall was used for characterization of precipitation if we have those, and in addition weekly and daily depends on processing power needed to fine grain such data.



heimo


________________________________
Von: Kathi Schleidt ***@***.***>
Gesendet: Dienstag, 27. Juni 2023 12:43
An: FAIRiCUBE/data-requests
Cc: Rainer Heimo; Mention
Betreff: Re: [FAIRiCUBE/data-requests] Climate data - Temperature (Issue #32)


@Mohinem<https://github.com/Mohinem> looking at the overview page, I see Temporal resolution: Hourly, could you please check this?

@robknapen<https://github.com/robknapen> @capoony<https://github.com/capoony> @sonjastndl<https://github.com/sonjastndl> @heimor<https://github.com/heimor>
looking at the inventory sheet, I see only 3 of the variables from this source requested (Temperature, Precipitation, Wind), when I look at the source description, I see more:

  *   Near-surface air temperature
  *   Near-surface specific humidity
  *   Near-surface wind speed
  *   Rainfall flux
  *   Snowfall flux
  *   Surface air pressure
  *   Surface downwelling longwave radiation
  *   Surface downwelling shortwave radiation

I'm assuming that the request for precipitation should actually refer to the Rainfall and Snowfall Flux variables. In the description is seems that these are derived from a more general ERA5 total precipitation source. Should we look for a more general "Precipitation" dataset, or are you OK with this split into Rain and Snow?

Also - the other variables also seem relevant, would you need these ingested?

**Comment by KathiSchleidt** [comment]
@Mohinem can you create the data requests for the additional climate variables available from the URI under investigation? Basically duplicates of this one, but for the following variables:
- Near-surface specific humidity
- Near-surface wind speed
- Rainfall flux
- Snowfall flux
- Surface air pressure
- Surface downwelling longwave radiation
- Surface downwelling shortwave radiation

You can modify the precipitation request for either rainfall or snowfall flux, create the rest so that we can track each concept individually.

Still unclear to me are the temporal intervals, both what's available as well as what's required.
- @Mohinem in the dataset description, it mentions hourly values, whereas you've stated that only monthly values are available. Could you please clarify? Just because the web site only lets you download in monthly blocks doesn't mean that the data isn't available on a daily level
- @heimor @robknapen 
  - what would be more advantageous for you, importing all temporal resolutions, or starting with coarser grained data for all datasets, then moving on to finer data?
  - in addition to cumulative values, there's mention of mean, stddev, min and max. Question is how we structure these individual values. I currently see 3 options
    - Bands: for each cell, we provide all 5 values, as bands. Not sure it will be possible to just request one band as a TIF, so for download you'd end up needing NetCDF
    - Dimension: creating a dimension for these 5 values. This would allow for retrieval as TIF by slicing only one of the 5
    - Individual Coverage: creating a separate coverage object for each of the 5 values. Advantage is you can access easily as TIF, disadvantage is that we end up with oodles of individual coverage objects

Thoughts?

**Comment by robknapen** [comment]
@KathiSchleidt Starting with months as temporal resolution is fine for our use case. I don’t expect that we will need hourly or daily weather data. One step finer would be dekadel values (10-day intervals), that is also common for agriculture and climate data.

Structure of the values: I am kind of expecting it to be 3 dimensional data, with the weather attributes as one of the dimensions (so your second option). How to select/slice from that cube and then encode it into a format such as tiff or netcdf is a processing step, or not?

**Comment by sonjastndl** [comment]
@KathiSchleidt For the Drosophila Use Case monthly data or weekly data is sufficient as well. For us, it is more important at the moemnt to have more variables (basically all variables mentioned for this dataset)  for the same time, so we can increase more power to our testing pipelines.

**Comment by KathiSchleidt** [comment]
@Mohinem looks like the votes are in, please start with monthly values. 

For the 5 values to be provided (average, mean, stddev, min and max), we'd like this as a (categorial) dimension, so one can either slice for one value and retrieve in a flat format (e.g. TIF), alternatively download all values as NetCDF.

**Comment by capoony** [comment]
@Mohinem, I do not find a data request issues for **near-surface air temperature**, which would be of central importance for our *Drosophila* Genomics  Use Case. 

We would specifically need the following layer resolution:

**Temporal resolution:**  We would need at least weekly resolution for the years 2011-2018. 
**Spatial resolution:**  The minimum spatial resolution encompasses all of Europe, at least from 35.3, -12.6 to 65.1, 45.9

How long will it putatively take for the ingestion? We need to plan our follow-up analyses accordingly.

Thanks a lot, 

Martin & Sonja

@KathiSchleidt, @heimor FYI

**Comment by Mohinem** [comment]
@capoony I will do this right now. I hope to complete it by tomorrow.

Brand new issue for the same -

https://github.com/FAIRiCUBE/data-requests/issues/50

The interesting thing is, the documentation mentions hourly values (https://cds.climate.copernicus.eu/cdsapp#!/dataset/derived-near-surface-meteorological-variables?tab=overview) but I can only download monthly values from the source (https://cds.climate.copernicus.eu/cdsapp#!/dataset/derived-near-surface-meteorological-variables?tab=form).

As of now, I will import monthly values.

**Comment by capoony** [comment]
Awesome, thanks a lot and many greetings!Am 01.08.2023 um 16:17 schrieb Mohit Kumar Basak ***@***.***>:﻿
@capoony I will do this right now. I hope to complete it by tomorrow.
Brand new issue for the same -
#50
The interesting thing is, the documentation mentions hourly values (https://cds.climate.copernicus.eu/cdsapp#!/dataset/derived-near-surface-meteorological-variables?tab=overview) but I can only download monthly values from the source (https://cds.climate.copernicus.eu/cdsapp#!/dataset/derived-near-surface-meteorological-variables?tab=form).
As of now, I will import monthly values.

**Comment by KathiSchleidt** [comment]
@Mohinem I just opened the data for May 2019 in QGIS. Time is provided as "hours since 1900-01-01", so Unix-Time taken down to hours. In the description, we also find the individual times (NETCDF_DIM_time_VALUES={1046016,1046017,1046018,...,1046759}). Based on the difference between start and end values, this nicely aligns with the 744 hrs in May (24hrs x 31days). In addition, we seem to have 744 bands in the NC file, so assuming that these are the individual hours.

**Comment by KathiSchleidt** [comment]
My bigger worry is the funky CRS used, Ellipsoid similar to but different to WGS84

When I plot this against OSM, everthing is shifted:

![grafik](https://github.com/FAIRiCUBE/data-requests/assets/11915304/3a0c5635-36f4-4bcc-900d-d2ac30a57f3d)

More on the funky Ellipsoid soon!!!

**Comment by Mohinem** [comment]
I am using python scripts to transform the data and generate TIFF files for each day, and it looks like the new generated raster file overlaps fully with OSM.
![image](https://github.com/FAIRiCUBE/data-requests/assets/13696328/d08f5a7b-e1a0-4a1a-880f-35013035df11)
I think the data makes sense for the air temperature (northern hemisphere is colder on 1st Feb, Himalayas is colder than rest of India, Andes is colder than rest of South America). One major difference is that the no data value is 0 (was previously a large value, will revert to it if that's highly required).

**Comment by KathiSchleidt** [comment]
@Mohinem definitely looks better than what showed in my QGIS, but I do see some misalignments (most visible is off the western coast of South America, data ranging out into the sea. Hard to say more at this resolution.) Could you provide the same image zoomed in to Europe?

On the no-data value, am I correct in assuming the temperatures are provided in Kelvin? Is there any way we could indicate this in the rangeType? Currently all we have to go on is:
- swe:field name="Gray"
- swe:Quantity definition="http://www.opengis.net/def/dataType/OGC/0/float32"
- swe:uom code="10^0"

Not really what we need.

In addition, do I understand correctly that you've now again shifted to ReferenceableGridByVectors in place of RectifiedGrid as AnsiDate also doesn't support provision of daily data? While I understand the issues with the provision of years (leap-year issues), I'd thought that days are straightforward (in the RectifiedGrid version, we had day as the offset. And, even if AnsiDate can only cleanly support seconds, days are pretty stable with 24*60*60=86400 seconds). 
Does this mean that all coverages will be changed to ReferenceableGridByVectors?

**Comment by KathiSchleidt** [comment]
@pebau still waiting on feedback after over a month. I'd like a justification for utilizing ReferenceableGridByVectors in the only dataset so far that could actually properly utilize AnsiDate

**Comment by pebau** [comment]
we do it that way, no problem. You CAN use it properly, do not spread false statements please!

**Comment by KathiSchleidt** [comment]
@pebau if I make false statements, please correct them! 
From my exploration of the AnsiDate CRS, the base unit is actually day, not second as you've previously erroneously implied. Please clarify

**Comment by pebau** [comment]
to continue work on this we obviously need more information ( as the above discussion shows). Someone needs to feed the inventory v2 table with the missing data.

@KathiSchleidt would that be (S4E, NHM had indicated prio 1) or you?

**Comment by KathiSchleidt** [comment]
@pebau please clearly specify what information you're still missing? 

In addition, please clarify as to your usage of AnsiDate, seems in clear contradiction with the specification

**Comment by pebau** [comment]
appreciated how all worked together to solve the question marks on the data - a nice showcase for the difficulties data wranglers encounter today!

For us to ingest (if data are still needed, at the last meeting the outcome was that all needed by the UCs is available as for now), please conflate all into the [standard data request information](https://github.com/FAIRiCUBE/data-requests/wiki/How-to-Add-Data); closing this.
